### ISSUE TITLE
Trainer's bug during adversarial training

### ISSUE THREAD

**Issue by dangne** [QData/TextAttack#498]
**Describe the bug**
Bug in [`File "textattack/trainer.py", line 213, in _generate_adversarial_examples`](https://github.com/QData/TextAttack/blob/d129e7b8b004b295fc5a26241636c6d9cc313a51/textattack/trainer.py#L213). Missing a column for the "adversarial_example" tag.

```
adversarial_examples = [
    (
        tuple(r.perturbed_result.attacked_text._text_input.values()),
        r.perturbed_result.ground_truth_output,
        "adversarial_example",
    )
    for r in results
    if isinstance(r, (SuccessfulAttackResult, MaximizedAttackResult))
]
```

The list of adversarial examples declared in `File "textattack/trainer.py", line 202, in _generate_adversarial_examples`
contains 3 columns:

1. The raw text
2. The ground truth label
3. An `"adversarial_example"` tag to differentiate it from clean examples.

However, line 213 misses a column for the adversarial tag, which leads to it being neglected by the [`collate_fn`](https://github.com/QData/TextAttack/blob/d129e7b8b004b295fc5a26241636c6d9cc313a51/textattack/trainer.py#L376) later in the code.  

**Expected behavior**
The adversarial tag should be retrieved and  the statement `if len(item) == 3` should be evaluated to `True`.

**Screenshots or Traceback**
![image](https://user-images.githubusercontent.com/36122564/127312275-0e91751f-cd63-49af-967b-db92231dd0d1.png)
I tried adversarial training with a dataset of a single example. The picture above is a batch containing a clean example and its corresponding adversarial example (successfully attacked). However, they were both marked as clean examples.

**System Information (please complete the following information):**
 - OS: Linux
 - Library versions: `pytorch==1.8.1`, `transformers==4.9.1`
 - Textattack version: `textattack==0.3.0`

**Comment by qiyanjun** [comment]
Thanks. Possible to submit a PR?

**Comment by dangne** [comment]
I'm working on it.
